frontend/src/actions/quayActions.js

```diff
@@ -267,6 +271,36 @@ export const fetchGraphData = (uuid) => async (dispatch, getState) => {
 
 export const tableReCalcValues = () => (dispatch, getState) => {
   const { page, perPage } = getState().quay;
-  dispatch(getQuaySummary());
+
   dispatch(setQuayPageOptions(page, perPage));
 };
+
+export const buildFilterData = () => async (dispatch, getState) => {
+  try {
+    const { tableFilters, categoryFilterValue } = getState().quay;
+
+    const params = dispatch(getRequestParams("quay"));
+
+    const response = await API.get("/api/v1/quay/filters", { params });
+    if (response.status === 200 && response?.data?.filterData?.length > 0) {
+      let data = cloneDeep(response.data.filterData);
+      for (let i = 0; i < tableFilters.length; i++) {
+        for (let j = 0; j < data.length; j++) {
+          if (tableFilters[i]["value"] === data[j]["key"]) {
+            data[j]["name"] = tableFilters[i]["name"];
+          }
+        }
+      }
```

Comment:
This logic confuses me. Right now I'm going to comment and pass it; later I'll make a second pass and actually look at the default store value, which might possibly shed some light. In general, though, it seems that if you need to do post-processing of your own new API on the UI side, maybe the API output could be designed differently to make that easier?

Reply:
Yeah, that could be done in the API output :roll_eyes: